AlphaWallet/Tokens/ViewControllers/TokensViewController.swift

```diff
@@ -280,32 +221,67 @@ class TokensViewController: UIViewController {
         configureSearchBarOnce()
     }
 
-    private func reload() {
-        isPromptBackupWalletViewHolderHidden = !(viewModel.shouldShowBackupPromptViewHolder && !promptBackupWalletViewHolder.subviews.isEmpty) || shouldHidePromptBackupWalletViewHolderBecauseSearchIsActive
-        tableView.reloadData()
-    }
-
     required init?(coder aDecoder: NSCoder) {
         return nil
     }
 
-    private func refreshView(viewModel: TokensViewModel) {
+    private func showOrHideBackupWalletViewHolder() {
+        isPromptBackupWalletViewHolderHidden = !(viewModel.shouldShowBackupPromptViewHolder && !promptBackupWalletViewHolder.subviews.isEmpty) || shouldHidePromptBackupWalletViewHolderBecauseSearchIsActive
+    }
+
+    private func bind(viewModel: TokensViewModel) {
+        navigationItem.largeTitleDisplayMode = viewModel.largeTitleDisplayMode
         view.backgroundColor = viewModel.backgroundColor
         tableView.backgroundColor = viewModel.backgroundColor
-    }
 
-    private func handleTokenCollectionUpdates() {
-        //NOTE: we don't apply .subscribe(on: DispatchQueue.main) for refresh table view immediatelly, otherwise an empty tokens view will be presented
-        tokenCollection.tokensViewModel
-            .sink { [weak self] viewModel in
-                guard let strongSelf = self else { return }
-                strongSelf.viewModel = viewModel
-                strongSelf.reload()
+        let input = TokensViewModelInput(
+            appear: appear.eraseToAnyPublisher(),
+            pullToRefresh: _pullToRefresh.eraseToAnyPublisher(),
+            selection: selection.eraseToAnyPublisher())
+
+        let output = viewModel.transform(input: input)
+
+        dataSource.numberOfRowsInSection.sink { [weak self, viewModel] section in
+            guard viewModel.sections[section] == .tokens || viewModel.sections[section] == .collectiblePairs else { return }
+            self?.handleTokensCountChange(rows: viewModel.numberOfItems(for: section))
+        }.store(in: &cancellable)
+
+        title = viewModel.walletDefaultTitle
+        
```

Comment:
Some spaces here in the otherwise empty line. Do you have "Automatically trim trailing whitespace" enabled in Xcode? I don't mind, but it'd introduce additional diffs in the future when editing the file.

Reply:
hah, yep, disabling it.

Reply:
It should be enabled :)

Reply:
i see, put it back ;)